todo-list/src/api/todo.ts

```diff
@@ -0,0 +1,31 @@
+import { Todo } from "../types/todos/todos.type";
+import { safeFetch } from "./safeFetch";
+
+export async function fetchTodosByUserId(userId: number): Promise<Todo[]> {
+  const url = `/todos/user/${userId}`;
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error("Failed to fetch todo");
+  }

```

Comment:
This code hide the true error response from the server, which can make debugging and error tracing harder.

Consider logging this error the console.